tests/e2e/specs/modules/search-console/admin-bar.test.js

```diff
@@ -0,0 +1,122 @@
+
+		await Promise.all( [
+			page.hover( '#wp-admin-bar-google-site-kit' ),
+			page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/data/' ) ),
+		] );
+
+		const adminBarApp = await page.$( '#js-googlesitekit-adminbar' );
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-data-block__title', { text: /total clicks/i } );
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-data-block__title', { text: /total impressions/i } );
+		// Ensure Analytics CTA is displayed
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-cta-link', { text: /Set up analytics/i } );
+		// More details link
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-cta-link', { text: /More details/i } );
+		await adminBarApp.dispose();
+	} );
+
+	it( 'loads when editing a post with data in Search Console', async () => {
+		const { searchConsole } = adminBarMockResponses;
+		// Data is requested when the Admin Bar app loads on first hover
+		mockBatchResponse = searchConsole;
+
+		// Navigate to edit view for this post
+		await Promise.all( [
+			expect( page ).toClick( '#wp-admin-bar-edit a', { text: /edit post/i } ),
+			page.waitForNavigation( { waitUntil: 'load' } ),
+		] );
+
+		// We're now in Gutenberg.
+
+		await Promise.all( [
+			page.hover( '#wp-admin-bar-google-site-kit' ),
+			page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/data/' ) ),
+		] );
+
+		await page.evaluate( () => {
+			// Temporarily replace XMLHttpRequest.send with a no-op to prevent a DOMException on navigation.
+			// https://github.com/WordPress/gutenberg/blob/d635ca96f8c5dbdc993f30b1f3a3a0b4359e3e2e/packages/editor/src/components/post-locked-modal/index.js#L114
+			window.XMLHttpRequest.prototype.send = function() {};
+		} );
+
+		const adminBarApp = await page.$( '#js-googlesitekit-adminbar' );
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-data-block__title', { text: /total clicks/i } );
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-data-block__title', { text: /total impressions/i } );
+		// Ensure Analytics CTA is displayed
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-cta-link', { text: /Set up analytics/i } );
+		// More details link
+		await expect( adminBarApp ).toMatchElement( '.googlesitekit-cta-link', { text: /More details/i } );
+		await adminBarApp.dispose();
+	} );
+
+	it( 'links "More details" to the dashboard details view for the current post', async () => {
+		const { searchConsole } = adminBarMockResponses;
+		// Data is requested when the Admin Bar app loads on first hover
+		mockBatchResponse = searchConsole;
+
+		await Promise.all( [
+			page.hover( '#wp-admin-bar-google-site-kit' ),
+			page.waitForResponse( ( res ) => res.url().match( 'google-site-kit/v1/data/' ) ),
+		] );
+
+		await expect( page ).toMatchElement(
+			'#js-googlesitekit-adminbar .googlesitekit-cta-link',
+			{ text: /More details/i, visible: true, timeout: 5000 }
```

Comment:
I take it the default 3000ms timeout was not enough?

Reply:
Let me try without this - this may have been one of the things we tried when debugging that I may have forgotten to back out.

Reply:
Trying this again, it passes for me but I think it's ok to keep. I would use `page.waitForSelector` (defaults to 30s timeout) here but it doesn't offer the same scoping that `expect-puppeteer` does.